### PR TITLE
Added STK atmosphere improvements

### PR DESCRIPTION
**Pull request by klingerj**
Applies alpha to the atmosphere that causes the atmosphere (as viewed from the ground) to disappear as the sun goes down if the lighting flag is enabled. Also alters alpha based on viewer position relative to the ground.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
@klingerj thanks for the pull request!

I noticed that [`CHANGES.md`](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CHANGES.md) has not been updated. If this change updates the Cesium API in any way, fixes a bug, or makes any non-trivial update, please add a bullet point to `CHANGES.md` and bump this pull request so we know it was updated. For more info, see the [Pull Request Guidelines]( https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#pull-request-guidelines).

__I am a bot who helps you make Cesium awesome!__ Thanks again.

**Comment by emackey** [comment]
Looks awesome!

Here's master, with the high-quality stars:

atmosphere_old

And here's this branch:

atmosphere_new

**Comment by pjcozzi** [comment]
At quick glance the code is OK with me.  @emackey do you want to review and merge?

File: Source/Shaders/SkyAtmosphereVS.glsl [hunk 1]
```diff
@@ -126,5 +127,6 @@ void main(void)
     v_mieColor = frontColor * KmESun;
     v_rayleighColor = frontColor * (InvWavelength * KrESun);
     v_toCamera = czm_viewerPositionWC - positionV3;
+    v_radii = u_cameraAndRadiiAndDynamicAtmosphereColor.zy;
```
Comment (emackey) [inline Source/Shaders/SkyAtmosphereVS.glsl]: Why pass this in as a varying, when the FS also has access to the same uniform?

/cc @abwood

**Comment by klingerj** [comment]
@emackey 'fraid I don't have a good answer as to why I did that...fixed it though. Thanks for pointing that out. @abwood This is going to have to change in STK too.

File: Source/Shaders/SkyAtmosphereFS.glsl [hunk 2]
```diff
@@ -73,5 +75,12 @@ void main (void)
     l = min(l, czm_luminance(rgb));
 #endif
 
-    gl_FragColor = vec4(rgb, min(smoothstep(0.0, 0.1, l), 1.0) * smoothstep(0.0, 1.0, czm_morphTime));
+    // Alter alpha based on how close the viewer is to the ground (1.0 = on ground, 0.0 = at edge of atmosphere)
+    float atmosphereAlpha = clamp((u_cameraAndRadiiAndDynamicAtmosphereColor.y - length(czm_viewerPositionWC)) / (u_cameraAndRadiiAndDynamicAtmosphereColor.y - u_cameraAndRadiiAndDynamicAtmosphereColor.z), 0.0, 1.0);
```
Comment (abwood) [inline Source/Shaders/SkyAtmosphereFS.glsl]: `length(czm_viewerPositionWC)` can be replaced by `u_cameraAndRadiiAndDynamicAtmosphereColor.x` (camera height)

Comment (klingerj) [inline Source/Shaders/SkyAtmosphereFS.glsl]: Done!

**Comment by abwood** [comment]
PR looks good to me too.  Awesome work @klingerj!

**Comment by pjcozzi** [comment]
Thanks again @klingerj!

**Pull request merged**